src/libraries/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -175,13 +169,25 @@ public override string ToString()
         /// </summary>
         internal static HttpMethod Normalize(HttpMethod method)
         {
-            // _http3EncodedBytes is only set for the singleton instances, so if it's not null,
+            // _http3Index is only set for the singleton instances, so if it's not null,
             // we can avoid the dictionary lookup.  Otherwise, look up the method instance in the
             // dictionary and return the normalized instance if it's found.
-            Debug.Assert(method != null);
-            return
-                method._http3EncodedBytes is null && s_knownMethods.TryGetValue(method, out HttpMethod? normalized) ? normalized :
-                method;
+            if (method._http3Index != null)
+                return method;
+
+            return method._method.ToUpperInvariant() switch
```

Comment:
This potentially allocates on every call.  It won't for the built-in instances, but it very well could for others.  This must not be merged.

Reply:
I thought it's a very uncommon but good point. Fixed the code